numbat-cli/src/main.rs

```diff
@@ -225,7 +276,7 @@ impl Cli {
     fn repl(&mut self) -> Result<()> {
         let interactive = std::io::stdin().is_terminal();
         let history_path = self.get_history_path()?;
-        let quiet_startup = self.args.quiet;
+        let intro_banner = self.args.intro_banner.unwrap_or(self.config.intro_banner);
```

Comment:
So what we do in other applications is the following: we load the config from the file (if there is any). And then we immediately *modify* that config struct with the available environment variables (which overwrite config options) and command line arguments (which overwrite env variables and config settings). This way, there is just one central place for configuration. And we could maybe get rid of `self.args`.